### PR TITLE
fix: add micro theme export and switcher example

### PR DESCRIPTION
**Pull request by luwes**
- upgrade to Media Chrome 0.18.5
- adds the `micro` Media Chrome theme as an export to the `@mux/mux-player` package via
`@mux/mux-player/themes/micro` for easy consumption.
- adds a theme switcher on the Next.js example page with the core themes
- adds a theme getter and setter to Mux player

### CODE REVIEW

**Review by gkatsev — APPROVED** [review]
LGTM.

**Comment by gkatsev** [comment]
Huh, weird that the yarn install fails in the test build.

**Comment by luwes** [comment]
@cjpillsbury any idea why the seek to live tests are failing with Media Chrome 0.18.5?

SCR-20230227-rvo

**Comment by gkatsev** [comment]
I saw that test failing last week when I was adding the cross browser testing support. I think it's some time timing issue and supposedly some of the new changes that is happening as part of the live edge m-e-ui stuff should help here too.

File: packages/mux-player/src/media-theme-mux.html [hunk 1]
```diff
@@ -263,6 +263,68 @@
       flex-grow: 1;
       font-size: 21px;
     }
+
+    /** Once we implement these CSS vars in Media Chrome below selectors can be deleted! */
```
Comment (cjpillsbury) [inline packages/mux-player/src/media-theme-mux.html:267]: 🎉

File: packages/mux-player/src/media-theme-mux.html [hunk 2]
```diff
@@ -447,9 +509,8 @@
     </template>
 
     <template partial="LiveButton">
-      <!-- NOTE: Keeping "seek-live" part for backwards compatibility for now -->
       <media-live-button
-        part="{{section ?? 'top'}} live seek-live button"
+        part="{{section ?? 'top'}} live button"
```
Comment (cjpillsbury) [inline packages/mux-player/src/media-theme-mux.html:513]: chore(blocker): Make sure the docs are appropriately updated to reflect this removal.

File: yarn.lock [hunk 3]
```diff
@@ -10595,10 +10595,10 @@ mdn-data@2.0.4:
   resolved "https://registry.yarnpkg.com/mdn-data/-/mdn-data-2.0.4.tgz#699b3c38ac6f1d728091a64650b65d388502fd5b"
   integrity sha512-iV3XNKw06j5Q7mi6h+9vbx23Tv7JkjEVgKHW4pimwyDGWm0OIQntJJ+u1C6mg6mK1EaTv42XQ7w76yuzH7M2cA==
 
-media-chrome@0.18.4:
-  version "0.18.4"
-  resolved "https://registry.yarnpkg.com/media-chrome/-/media-chrome-0.18.4.tgz#50438f0635d9027a5d23c69e8baf75358b22dfcd"
-  integrity sha512-ftG68jxSUraibyjOhyNEueEW94jVA87wJRYPylS8b/k6/nm07yURRPmk0O321NaI+eQ6i15wpA8UFgKSoBOqnQ==
+media-chrome@0.18.5:
```
Comment (cjpillsbury) [inline yarn.lock:10598]: suggestion(non-blocker): Even though it's out of scope, mind also version bumping upchunk? I just cut a release yesterday, so might as well update it while we're in here.

File: packages/mux-player/package.json [hunk 4]
```diff
@@ -8,9 +8,12 @@
   "unpkg": "./dist/mux-player.js",
   "jsdelivr": "./dist/mux-player.js",
   "exports": {
-    "import": "./dist/index.mjs",
-    "require": "./dist/index.cjs.js",
-    "default": "./dist/index.cjs.js"
+    ".": {
```
Comment (cjpillsbury) [inline packages/mux-player/package.json:11]: question(blocker): Have you confirmed this doesn't break any of the build tools used by the "out of box" frameworks we target? Minimally, let's make sure all of our workspace example apps still work as expected before merge, and let's have as a required followup task confirming that no web build check builds fail after the canary release.

File: packages/mux-player/package.json [hunk 5]
```diff
@@ -8,9 +8,12 @@
...
+      "import": "./dist/index.mjs",
+      "require": "./dist/index.cjs.js",
+      "default": "./dist/index.cjs.js"
+    },
+    "./themes/micro": "./themes/micro.js"
```
Comment (cjpillsbury) [inline packages/mux-player/package.json:16]: praise: 🎉

File: packages/mux-player/src/index.ts [hunk 6]
```diff
@@ -605,6 +606,20 @@ class MuxPlayerElement extends VideoApiElement {
     return this.media?.mux;
   }
 
+  /**
+   * Gets the theme.
+   */
+  get theme() {
```
Comment (cjpillsbury) [inline packages/mux-player/src/index.ts:612]: praise: 😎

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
a couple of asks, but overall this is exciting!

**Comment by cjpillsbury** [comment]
> @cjpillsbury any idea why the seek to live tests are failing with Media Chrome 0.18.5?
> 

@luwes Since migrating to the media chrome impl for seek live controls/behavior, these tests are more unreliable. This _**should**_ be improved after the media-ui-extensions migration is completed, as we'll be using stream-derived values for the live edge computation.

File: packages/mux-player/src/media-theme-mux.html (hunk 2 above)
Comment (luwes) [reply packages/mux-player/src/media-theme-mux.html:513]: added https://github.com/muxinc/docs.mux.com/pull/856

File: packages/mux-player/package.json (hunk 4 above)
Comment (luwes) [reply packages/mux-player/package.json:11]: just checked the examples, they all build and the player loads and plays

File: packages/mux-player/src/template.ts [hunk 7]
```diff
@@ -78,7 +78,7 @@ export const content = (props: MuxTemplateProps) => html`
       player-software-name="${props.playerSoftwareName ?? false}"
       player-software-version="${props.playerSoftwareVersion ?? false}"
       env-key="${props.envKey ?? false}"
-      stream-type="${props.streamType ?? false}"
+      stream-type="${props.streamType ? (isLiveOrDVR(props) ? 'live' : 'on-demand') : false}"
```
Comment (luwes) [inline packages/mux-player/src/template.ts:81]: @cjpillsbury fixed the issue causing the tests to fail here for now

Comment (cjpillsbury) [reply packages/mux-player/src/template.ts:81]: praise: Good catch here.

**Review by cjpillsbury — APPROVED** [review]
awesome!

**Pull request merged**